modules/OTAR_CRISPR.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python3
+
+import argparse
+from datetime import datetime
+import gzip
+import json
+import logging
+import sys
+
+import pandas as pd
+
+# The statisticalTestTail is inferred by the column name wich is being filtered on:
+FILTER_COLUMN_MAP = {
+    'pos|fdr': 'upper tail',
+    'neg|fdr': 'lower tail',
+    'neg|p-value': 'lower tail',
+    'pos|p-value': 'upper tail'
+}
+
+class OTAR_CRISPR_study_parser(object):
+    def __init__(self, study_file: str) -> None:
+
+        self.study_file = study_file
+
+        # Read and process study table:
+        study_df = pd.read_csv(study_file, sep='\t', skiprows=[1])
+
+        self.study_df = (
+            study_df
+
+            # drop rows with no study id or data file:
+            .loc[study_df.studyId.notna() & study_df.dataFile.notna()]
+            .assign(
+                # Split mapped diseases to list:
+                diseaseFromSourceMappedId=study_df.diseases.str.replace(' ', '').str.split('|'),
```

Comment:
I am blown away by this use of `assign` and how it can replicate the `withColumn` function 😍
I have one question: when you refer to `study_df.diseases`, is this the whole initial column or the filtered one (after loc)?
In the Pandas documentation I see this could also be an alternative:
```
diseaseFromSourceMappedId=lambda X: X.diseases.str.replace(' ', '').str.split('|'),
```

Reply:
Yes. That's correct, however it's resolved by using the available index. You can see how it works in this example:

```python
import pandas as pd

df = pd.DataFrame([
    {'a': 2, 'b': 24},
    {'a': 4, 'b': None},
    {'a': 12, 'b': 100}
])

(
    df
    .loc[df.b.notna()]
    .assign(
        double=df.a*2
    )
)
```

Reply:
But in general I try to follow the canonical method using the `lambda`.